packages/logger/src/index.ts

```diff
@@ -0,0 +1,216 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/**
+ * @packageDocumentation
+ *
+ * A logger for fuel based on the [debug](https://www.npmjs.com/package/debug) module.
```

Comment:
Should we have docs for this, or should this be used internally only?

Reply:
My intention was for this is to be an internal tool that is used primarily by us which is why I hadn't documented it ( I explained in more detail [here](https://github.com/FuelLabs/fuels-ts/pull/2366#discussion_r1643367640) ) 

If we think developers building on TS SDK will use this tool then we can document it. My concern is just that we don't want to have the TS docs be too lengthy.

<img width="249" alt="Screenshot 2024-06-20 at 10 25 49 AM" src="https://github.com/FuelLabs/fuels-ts/assets/9755286/3fc4a4d3-de0e-4b40-9d1b-435b3469610f">

Reply:
It should probably go into `Utils` so it wouldn't affect the first-level navigation, but I think we should first use it ourselves and then, when we're happy with it, expose it in the docs if we think it makes sense.

Reply:
I agree @nedsalk

Reply:
Makes sense.